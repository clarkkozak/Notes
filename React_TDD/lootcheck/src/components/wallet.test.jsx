import React from 'react'
import { shallow } from 'enzyme'
import { Wallet } from './wallet'

describe('Wallet', () => {
  const props = { balance: 20 } 
  const wallet = shallow(<Wallet {...props} />)
  it('Renders Properly', () => {
    expect(wallet).toMatchSnapshot()
  })

  it('displays the balance from props', () => {
    expect(wallet.find('.balance').text()).toEqual('Wallet balance: 20')
  })

  it('creates an input to deposit or withdraw from the balance', () => {
    expect(wallet.find('.input-wallet').exists()).toBe(true)
  })

  describe('when a user types into the wallet input', () => {
    const userBalance = '25'

    beforeEach(() => {
      wallet.find('.input-wallet').simulate('change', { target: { value: userBalance }})
    })

    it('updates the local wallet balance in `state` and coverts it to a number', () => {
      expect(wallet.state().balance).toEqual(parseInt(userBalance, 10))
    })

  })
  

})