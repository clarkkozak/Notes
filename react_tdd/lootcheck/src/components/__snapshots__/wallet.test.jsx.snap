// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet Renders Properly 1`] = `ShallowWrapper {}`;
